sw/device/lib/testing/test_framework/BUILD

```diff
@@ -44,3 +44,79 @@ cc_library(
         "//sw/device/lib/runtime:print",
     ],
 )
+
+# TODO(lowRISC/opentitan#7972): clean up check.h after completing the
+# migration to bazel.
+cc_library(
+    name = "check",
+    hdrs = [
+        "test_status.h",
+        "//sw/device/lib/testing:check",
+    ],
+    visibility = ["//visibility:private"],
+)
+
+cc_library(
+    name = "freertos_config",
+    hdrs = [
+        "FreeRTOSConfig.h",
+    ],
+    includes = ["."],
+    deps = [
+        "//sw/device/lib/arch:device",
+    ],
+)
+
+cc_library(
+    name = "freertos_port",
+    srcs = [
+        "freertos_hooks.c",
+        "freertos_port.S",
+        "freertos_port.c",
+    ],
+    hdrs = [
+        "ottf_macros.h",
+    ],
+    deps = [
+        ":check",
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/lib:irq",
+        "//sw/device/lib/base",
+        "//sw/device/lib/dif:rv_timer",
+        "//sw/device/lib/dif:uart",
+        "//sw/device/lib/runtime:hart",
+        "//sw/vendor/freertos_freertos_kernel:kernel",
+    ],
+)
+
+cc_library(
+    name = "ottf",
+    srcs = [
+        "ottf.c",
+        "ottf_isrs.S",
+        "ottf_isrs.c",
```

Comment:
In #9335, on the meson side of things, I decoupled the ISRs from the main ottf lib (and just included them as a dep) in order to allow these ISRs to be used in test code that doesn't use the OTTF but still makes use of the test 2nd boot stage asm init code (`flash_crt.S`) that registers the same interrupt vectors (i.e. `sw/device/riscv_compliance_support/` and `sw/device/benchmarks/coremark/`). The goal was to eventually deprecate the `handler.<h,c>` code so we only have to maintain one set of test ISRs.  Not sure if you had a different idea in mind for this?

Reply:
The old IRQ handlers have now been removed (#9575), so these (`ottf_isrs.<S,h,c>` will need to be build as a separate lib (or be lumped in with the `//sw/device/lib:irq` lib as was previously the case). Also, as we discussed offline, I accidentally removed the entire `//sw/device/lib:irq` build target, and that will need to be added back in (to at least contain the IRQ enable/disable functions), sorry about that!